pkg/controllers/disruption/controller.go

```diff
@@ -265,6 +265,34 @@ func (c *Controller) createReplacementNodeClaims(ctx context.Context, m Method,
 	return nodeClaimNames, nil
 }
 
+func (c *Controller) MarkDisrupted(ctx context.Context, m Method, candidates ...*Candidate) error {
+	stateNodes := lo.Map(candidates, func(c *Candidate, _ int) *state.StateNode {
+		return c.StateNode
+	})
+	if err := state.RequireNoScheduleTaint(ctx, c.kubeClient, true, stateNodes...); err != nil {
+		return fmt.Errorf("tainting nodes with %s: %w", pretty.Taint(v1.DisruptedNoScheduleTaint), err)
+	}
+
+	providerIDs := lo.Map(candidates, func(c *Candidate, _ int) string { return c.ProviderID() })
+	c.cluster.MarkForDeletion(providerIDs...)
+
+	err := multierr.Combine(lo.Map(candidates, func(candidate *Candidate, _ int) error {
+		// refresh nodeclaim before updating status
+		nodeClaim := &v1.NodeClaim{}
+		err := c.kubeClient.Get(ctx, client.ObjectKeyFromObject(candidate.NodeClaim), nodeClaim)
+		if err != nil {
+			return err
+		}
+		nodeClaim.StatusConditions().SetTrueWithReason(v1.ConditionTypeDisruptionReason, v1.ConditionTypeDisruptionReason, string(m.Reason()))
+		return c.kubeClient.Status().Update(ctx, nodeClaim)
+	})...)
+	if err != nil {
+		c.cluster.UnmarkForDeletion(providerIDs...)
+		return fmt.Errorf("updating nodeclaim status: %w", err)
+	}
```

Comment:
I think you can remove this as part of the beginning of the disruption reconcile loop